airflow/utils/context.py

```diff
@@ -124,6 +142,11 @@ def __reduce_ex__(self, protocol: int) -> Tuple[Any, ...]:
         items = [(key, self[key]) for key in self._context]
         return dict, (items,)
 
+    def __copy__(self) -> "Context":
+        new = type(self)(copy.copy(self._context))
+        new._deprecation_replacements = self._deprecation_replacements.copy()
+        return new
+
     def __getitem__(self, key: str) -> Any:
         with contextlib.suppress(KeyError):
             warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]), stacklevel=2)
```

Comment:
```suggestion
            warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))
```

This is the culprit, as it issues the warning from `jinja2/runtime.py:154` instead.

Reply:
Thanks! I’m not quite sure why Jinja2 seemingly made one call stack disappear, but it is what it is.